service/src/main/java/com/codeforcommunity/processor/InactiveUserProcessorImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.codeforcommunity.processor;
+
+import com.codeforcommunity.api.IInactiveUserProcessor;
+import com.codeforcommunity.requester.Emailer;
+import io.vertx.core.Vertx;
+import org.jooq.DSLContext;
+import org.jooq.Result;
+import org.jooq.generated.tables.records.AdoptedSitesRecord;
+import org.jooq.generated.tables.records.StewardshipRecord;
+import org.jooq.generated.tables.records.UsersRecord;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.jooq.generated.Tables.*;
+
+public class InactiveUserProcessorImpl implements IInactiveUserProcessor {
+  private final Emailer emailer;
+  private final DSLContext db;
+  private final List<Integer> emailedIds;
+  private static final int PERIOD = 21;
+
+  public InactiveUserProcessorImpl(DSLContext db, Emailer emailer) {
+    this.emailer = emailer;
+    this.db = db;
+    this.emailedIds = new ArrayList<>();
+
+    Vertx vertx = Vertx.vertx();
+
+
+    //Note: First check is 24h after starting
+    long timerId = vertx.setPeriodic(TimeUnit.DAYS.toMillis(1), id -> {
+      this.emailInactiveUsers();
+    });
+  }
+
+
+  //Query users who haven't performed activity in the given range
+  private List<UsersRecord> getInactiveUsers(int range) {
+    Calendar cal = new GregorianCalendar();
+    cal.setTime(new Timestamp(System.currentTimeMillis()));
+    Date currentDate = new Date(cal.getTime().getTime());
+    cal.add(Calendar.DAY_OF_MONTH, -1 * range); //21 days ago
+    Date floorDate = new Date(cal.getTime().getTime());
+    Result<UsersRecord> users = db.selectFrom(USERS).fetch();
+    List<UsersRecord> inactiveUsers = new ArrayList<>();
+    for (UsersRecord user : users) {
```

Comment:
If a person has more than one site adopted, do we want to send them one email for each site? I think it would be better if we compiled each site that a user has not checked into one email.

Reply:
Moved to ProtectedSiteProcessorImpl 👍

Reply:
Currently it sends one email per site, but only once those sites reach their limit. The one question I would have about compiling multiple into one email is when is this email sent? When the first site is unattended or the last? I can keep the same sending schedule but remind users of all of their inactive sites if that would be easier.